snack-lib/build.nix

```diff
@@ -114,23 +117,24 @@ rec {
                 modSpec.moduleFiles
             ) >= 1
         ) base;
-    in stdenv.mkDerivation
-    { name = objectName;
       src = symlinkJoin
         { name = "extra-files";
           paths = [ extraFiles ] ++ modSpec.moduleDirectories;
         };
-      phases =
-        [ "unpackPhase" "buildPhase" ];
-
+    in builtins.derivation
+    { name = objectName;
+      system = stdenv.system;
       imports = map (mmm: mmm.moduleName) modSpec.moduleImports;
-      buildPhase =
+      PATH = lib.makeBinPath [ coreutils rsync ghc ];
+      builder = writeScript "${objectName}-builder"
```

Comment:
One thing you can do is use the `passAsFile` feature to avoid creating the other derivation issues from the `writeScript` call and make this even faster. Also if the script is POSIX-shell compatible, "/bin/sh" is already part of all of the sandboxes so bash can also be skipped.

Not tested but it would look something like this:

```
derivation {
  name = "xxx";
  system = stdenv.system;
  buildCommand = ''
    # your script here
  '';
  passAsFile = [ "buildCommand" ];
  builder = "/bin/sh";
  args = ["-c", ". $buildCommandPath"];
}
```
}
```